File: common/src/main/java/guepardoapps/lucahome/common/interfaces/classes/ILucaClass.java
package guepardoapps.lucahome.common.interfaces.classes;

import android.support.annotation.NonNull;

public interface ILucaClass {
    enum LucaServerDbAction {Null, Add, Update, Delete}

    int GetId();

    void SetIsOnServer(boolean isOnServer);

    boolean GetIsOnServer();

    void SetServerDbAction(@NonNull LucaServerDbAction serverDbAction);

    LucaServerDbAction GetServerDbAction();

    String CommandAdd() throws NoSuchMethodException;

    String CommandUpdate() throws NoSuchMethodException;

    String CommandDelete() throws NoSuchMethodException;
}

File: common/src/main/java/guepardoapps/lucahome/common/models/ILucaClass.kt
package guepardoapps.lucahome.common.models

import guepardoapps.lucahome.common.enums.ServerDatabaseAction

import java.io.Serializable
import java.util.UUID

interface ILucaClass : Serializable {
    val uuid: UUID
    var roomUuid: UUID

    val commandAdd: String
    val commandUpdate: String
    val commandDelete: String

    var isOnServer: Boolean
    var serverDatabaseAction: ServerDatabaseAction
}
